blindsign/blindrsa/rsa.go

```diff
@@ -87,12 +83,17 @@ func encrypt(c *big.Int, pub *rsa.PublicKey, m *big.Int) *big.Int {
 	return c
 }
 
+// decrypt performs an RSA decryption, resulting in a plaintext integer. If a
+// random source is given, RSA blinding is used.
 func decrypt(random io.Reader, priv *rsa.PrivateKey, c *big.Int) (m *big.Int, err error) {
 	// TODO(agl): can we get away with reusing blinds?
 	if c.Cmp(priv.N) > 0 {
 		err = rsa.ErrDecryption
 		return
 	}
+	if priv.N.Sign() == 0 {
+		return nil, rsa.ErrDecryption
+	}
 
 	var ir *big.Int
 	if random != nil {
```

Comment:
Are we missing this line? https://cs.opensource.google/go/go/+/master:src/crypto/rsa/rsa.go;l=558

Reply:
Doesn't seem too important.

Reply:
it shuffles the random source in case a user-provided source is provided.
in this package, the random source is always `crypto/rand`, which it's assumed to be secure.